### PR TITLE
Cover both width and height for the sidebar preview

### LINKED ISSUE #19369: Blurry previews in the right side-bar
### Steps to reproduce

1. Set max preview at 756x756
2. Upload an image which is bigger. Mine was 800x600
3. Bring up the details view

### Expected result 

A crisp preview filling up the space

### Actual result

A blurry preview filling up the space

The container is about 430x240
The image is 320x240, stretched to fit

@jancborchardt @icewind1991

### PR DESCRIPTION
**Pull request by icewind1991**
Request a preview that fills the full with and height of the sidebar preview.

Fixes #19369

cc @jancborchardt @oparoz

### CODE REVIEW

**Comment by oparoz** [comment]
Thanks. It's much better.
I saw one regression and one bug.

* When looking at the details of an image in portrait mode, there is a large gap between the bottom of the image and the title
* When resizing the browser, the image shrinks, but not the height of the container

**Comment by oparoz** [comment]
And something else. Are images blown up using the config settings which allows scaling?

**Comment by icewind1991** [comment]
Portrait images (and their container) should now be sized correctly also when resizing the browser

> Are small images blown up using the config settings which allows scaling?

I dont think so

**Comment by oparoz** [comment]
>> Are small images blown up using the config settings which allows scaling?

> I dont think so

Might be a bug then because small images are shown larger than their original size.

**Comment by oparoz** [comment]
Some screenshots for people not willing to switch branch.

### Small, square
square

### Tall
tall

### Taller
supertall

### Tiny, blown up
blown_up

**Comment by jancborchardt** [comment]
Please review @owncloud/designers, I’m on vacation for 2 weeks.

**Comment by icewind1991** [comment]
Fixed small previews being upscaled and improved handling of images with an aspect ratio > 16/9

**Comment by MorrisJobke** [comment]
Tested and looks good :+1:

**Comment by MorrisJobke** [comment]
@icewind1991 Preview tests fail - please fix them

File: lib/private/preview.php [hunk 1]
```diff
@@ -441,6 +441,7 @@ public function deleteAllPreviews() {
 	 * @return string|false path to the cached preview if it exists or false
 	 */
 	public function isCached($fileId) {
+		return false;
```
Comment (oparoz) [inline lib/private/preview.php]: Test probably fail because if this leftover debugging  code

**Comment by MorrisJobke** [comment]
Rebased and fixed the issue that was pointed out by @oparoz

**Comment by oparoz** [comment]
Tests are now failing because the code breaks the max preview generation.

**Comment by icewind1991** [comment]
Ughh, failures seem to be random...

**Comment by icewind1991** [comment]
Since the input data for the tests is random...

Please dont do random data in tests, makes debuging a PITA

**Comment by oparoz** [comment]
The point of using random data is to emulate user data, otherwise people end up cheating to make the tests pass.

Looking at the failed tests, it seems that since you've removed the check for the aspect ratio when looking for a cached image, the cache doesn't return the max preview any more if the asked size is larger than the max size, which means that previews get generated over and over.

**Comment by oparoz** [comment]
When working on this, I also used some static arrays in order to always trigger the same case.

```
return [
			[0, -rand(200, 700), -rand(200, 700)],
			[0, rand(200, 700), rand(200, 700)],
			[0, -rand(200, 700), rand(200, 700)],
			[0, rand(200, 700), -rand(200, 700)],
			[0, -rand(200, 700), -rand(200, 700), true],
			[0, rand(200, 700), rand(200, 700), true],
			[0, -rand(200, 700), rand(200, 700), true],
			[0, rand(200, 700), -rand(200, 700), true],
			[0, -rand(200, 700), -rand(200, 700), false, true],
			[0, rand(200, 700), rand(200, 700), false, true],
			[0, -rand(200, 700), rand(200, 700), false, true],
			[0, rand(200, 700), -rand(200, 700), false, true],
			[0, -rand(200, 700), -rand(200, 700), true, true],
			[0, rand(200, 700), rand(200, 700), true, true],
			[0, -rand(200, 700), rand(200, 700), true, true],
			[0, rand(200, 700), -rand(200, 700), true, true],
			[1, -rand(200, 700), -rand(200, 700)],
			[1, rand(200, 700), rand(200, 700)],
			[1, -rand(200, 700), rand(200, 700)],
			[1, rand(200, 700), -rand(200, 700)],
			[1, -rand(200, 700), -rand(200, 700), true],
			[1, rand(200, 700), rand(200, 700), true],
			[1, -rand(200, 700), rand(200, 700), true],
			[1, rand(200, 700), -rand(200, 700), true],
			[1, -rand(200, 700), -rand(200, 700), false, true],
			[1, rand(200, 700), rand(200, 700), false, true],
			[1, -rand(200, 700), rand(200, 700), false, true],
			[1, rand(200, 700), -rand(200, 700), false, true],
			[1, -rand(200, 700), -rand(200, 700), true, true],
			[1, rand(200, 700), rand(200, 700), true, true],
			[1, -rand(200, 700), rand(200, 700), true, true],
			[1, rand(200, 700), -rand(200, 700), true, true],
		];
```

**Comment by PVince81** [comment]
Tests need to be deterministic. I suggest to pick a few random values manually, ones that are representative of the cases you want to cover, then hard-code them in the test array so they stay constant.

**Comment by icewind1991** [comment]
Should work now

**Comment by oparoz** [comment]
- [x] Fixes the original problem with blurry previews
- [x] Keeps small images with scaling defined in config
- [x] No side-effects
- [x] Tested

Just one note regarding the samples I posted above. Nothing has changed for tall previews, but that works for me.

:+1:

Possible enhancements for 9.0
* Resize previews when bowser window is resized
* Crop very tall pictures

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Pull request merged**